src/components/Calendar/day.js

```diff
@@ -31,18 +51,29 @@ function DayComponent(props) {
 
     return (
         <StyledDay role="gridcell">
-            <StyledDayButton
-                ref={buttonRef}
-                tabIndex={tabIndex}
-                onClick={() => onChange(new Date(date))}
-                isSelected={isSelected}
-                data-selected={isSelected}
-                onKeyDown={privateKeyDown}
-                onFocus={privateOnFocus}
-                onBlur={privateOnBlur}
+            <StyledRangeHighlight
+                isVisible={isWithinRange && !(isRangeStartDate && isRangeEndDate)}
+                isFirstInRange={isRangeStartDate}
+                isLastInRange={isRangeEndDate}
+                isFirstDayOfWeek={isFirstDayOfWeek}
+                isLastDayOfWeek={isLastDayOfWeek}
             >
-                {day}
-            </StyledDayButton>
+                <StyledDayButton
+                    ref={buttonRef}
+                    tabIndex={tabIndex}
+                    onClick={() => onChange(new Date(date))}
+                    onMouseEnter={() => privateOnHover(new Date(date))}
```

Comment:
why we need to create a new instance here, if we need to avoid date mutation the clone should happen inside the privateOnHover or onChange